harvest/storage/base_storage.py

```diff
@@ -30,6 +30,8 @@ def __init__(self):
         """
         self.storage_lock = Lock()
         self.storage = {}
+        self.N = 0
```

Comment:
If `N` is set to zero then wouldn't this clear the entire cache? Also should we make limit size and N inputs to the storage so users can customize it.